doc/verilog_systemverilog.txt

```diff
@@ -448,6 +448,19 @@ Example:
     Note: The commands |:VerilogFoldingAdd| and |:VerilogFoldingRemove| are
     provided to allow an easier management of this variable.
 
+                                                      *g:verilog_quick_syntax*
+g:verilog_quick_syntax~
+Default: undefined
+
+    Whena enabled, syntax regions will not be defined with Vim's 'syntax'
+    command. This can help performance if you are using an old machine or
+    veiwing a large file such as a netlist. This can be useful if you don't
+    care about automatic indentation but still wish to have syntax coloring.
```

Comment:
Hmm. I think if we want the plugin to me more advanced, it'll have to get slower. I think SystemVerilog's complex and inconsistent syntax is pushing Vim's syntax engine to it's limits.

For me personally, the automatic indent is incredibly valuable and saves me huge amounts of time so my preference will be going the route of making it smarter. My new indent script works without syntax regions but it is much slower than the current, and using syntax regions makes it much quicker than the current.

| Plugin | Syntax speed | Syntax speed (with folding) | Indent speed | Indent accuracy | Code Maintainability |
| ------ | ----------- | ---------------------------- | ------------ | --------------- | -------------------- |
| Current | Fast | Slow | OK | OK | Difficult |
| New (with syntax regions) |  Slow | Slow | Fast | Best | Easier (hopefully) |
| New (without syntax regions) |  Fast | Slow | Slow | Best | Easier (hopefully) |

I can definitely understand why you don't want to make the syntax slower. We can go several routes here:

1. Just go ahead with the changes and let the plugin get slower and smarter.
2. Keep the plugin as it is a start a new plugin with all new changes.
3. Incorporate everything into this plugin, i.e. have both old a new implementations and configure everything with variables. - This is my least favourite option.